h2/src/main/org/h2/value/ValueStringFixed.java

```diff
@@ -42,7 +70,14 @@ public int getType() {
      * @return the value
      */
     public static ValueStringFixed get(String s) {
-        s = trimRight(s);
+        return get(s, PRECISION_TRIM, null);
+    }
+    public static ValueStringFixed get(String s, int precision, Mode mode) {
+        if (mode != null && mode.padFixedStrings && precision < Integer.MAX_VALUE) {
```

Comment:
whats with the 
   precision < Integer.MAX_VALUE
will either be true, or indicate a bug?